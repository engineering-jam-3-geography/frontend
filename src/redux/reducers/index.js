import {combineReducers} from 'redux';
import modalsReducer from './modals';

const rootReducer = combineReducers({
    modals: modalsReducer,
});

export default rootReducer;
