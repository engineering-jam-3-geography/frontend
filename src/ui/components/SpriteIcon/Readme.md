SpriteIcon

    <SpriteIcon iconLink="#ico-about-us" />

SpriteIcon with size

    <SpriteIcon width="50" height="50" iconLink="#ico-about-us" />

SpriteIcon with className

    <SpriteIcon width="50" height="50" iconLink="#ico-about-us" className="icon" />
