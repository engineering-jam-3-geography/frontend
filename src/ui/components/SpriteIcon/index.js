import React from 'react';
import PropTypes from 'prop-types';

/**
 * Renders SpriteIcon component
 * @param {Object}
 * @returns {JSX}
 */
const SpriteIcon = ({
    width,
    height,
    className,
    focusable,
    iconLink
}) => (
    <svg width={width} height={height} className={className} focusable={focusable.toString()}>
        <use xlinkHref={iconLink} />
    </svg>
);

SpriteIcon.propTypes = {
    width: PropTypes.string,
    height: PropTypes.string,
    className: PropTypes.string,
    focusable: PropTypes.bool,
    iconLink: PropTypes.string
};

SpriteIcon.defaultProps = {
    focusable: false
};

export default SpriteIcon;
