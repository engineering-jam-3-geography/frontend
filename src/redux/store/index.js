import {createStore, applyMiddleware} from 'redux';
import {createBrowserHistory} from 'history';
import {connectRouter, routerMiddleware} from 'connected-react-router';
import thunk from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';
import {createLogger} from 'redux-logger';
import rootReducer from '../reducers';
import config from '../../config';

const {basename = ''} = config;

const history = createBrowserHistory({
    basename
});
const router = routerMiddleware(history);
const middleware = [router, thunk];

if (process.env.NODE_ENV === 'development') {
    const logger = createLogger({collapsed: true, diff: true});

    middleware.push(logger);
}

const store = createStore(
    connectRouter(history)(rootReducer),
    composeWithDevTools(
        applyMiddleware(...middleware)
    )
);

export {history};
export default store;
