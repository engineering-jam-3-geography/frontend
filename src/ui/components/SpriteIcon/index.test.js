import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import {compose} from 'redux';
import SpriteIcon from '.';

Enzyme.configure({adapter: new Adapter()});

describe('<SpriteIcon /> UI Component', () => {
    const shallowExpect = compose(expect, toJSON, shallow);

    test('SpriteIcon.', () => {
        shallowExpect(
            <SpriteIcon iconLink="#ico-about-us" />
        ).toMatchSnapshot();
    });

    test('SpriteIcon with size.', () => {
        shallowExpect(
            <SpriteIcon width="50" height="50" iconLink="#ico-about-us" />
        ).toMatchSnapshot();
    });

    test('SpriteIcon with className.', () => {
        shallowExpect(
            <SpriteIcon width="50" height="50" iconLink="#ico-about-us" className="icon" />
        ).toMatchSnapshot();
    });
});
