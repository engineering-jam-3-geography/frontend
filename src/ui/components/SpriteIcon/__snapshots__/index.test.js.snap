// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpriteIcon /> UI Component SpriteIcon with className. 1`] = `
<svg
  className="icon"
  focusable="false"
  height="50"
  width="50"
>
  <use
    xlinkHref="#ico-about-us"
  />
</svg>
`;

exports[`<SpriteIcon /> UI Component SpriteIcon with size. 1`] = `
<svg
  focusable="false"
  height="50"
  width="50"
>
  <use
    xlinkHref="#ico-about-us"
  />
</svg>
`;

exports[`<SpriteIcon /> UI Component SpriteIcon. 1`] = `
<svg
  focusable="false"
>
  <use
    xlinkHref="#ico-about-us"
  />
</svg>
`;
